src/main/java/io/swagger/swaggerhub/plugin/services/DefinitionFileFormat.java

```diff
@@ -4,21 +4,50 @@
 import io.swagger.util.Json;
 import io.swagger.util.Yaml;
 
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
 /**
  * Accepted file formats for API definition upload
  */
 public enum DefinitionFileFormat {
 
-    JSON(Json.mapper()),
-    YAML(Yaml.mapper());
+    JSON(Json.mapper(), Arrays.asList("json"), "json"),
+    YAML(Yaml.mapper(), Arrays.asList("yaml", "yml"), "yaml");
 
-    private final ObjectMapper mapper;
+    private final String fileFormat;
+    private ObjectMapper mapper;
+    private List<String> supportedFileExtensions;
 
-    DefinitionFileFormat(ObjectMapper mapper) {
+    DefinitionFileFormat(ObjectMapper mapper, List<String> supportedFileExtensions, String fileFormat) {
         this.mapper = mapper;
+        this.supportedFileExtensions = supportedFileExtensions;
+        this.fileFormat = fileFormat;
     }
 
     public ObjectMapper getMapper() {
         return mapper;
     }
+
+    public String getFileFormat() {
+        return fileFormat;
+    }
+
+    public List<String> getSupportedFileExtensions() {
+        return supportedFileExtensions;
+    }
+
+    public static Optional<DefinitionFileFormat> getByFileExtensionType(String value){
+        return EnumSet.allOf(DefinitionFileFormat.class).stream()
+                .filter(isFileExtensionSupportedForFileFormat(value))
+                .findFirst();
+    }
+
+    private static Predicate<DefinitionFileFormat> isFileExtensionSupportedForFileFormat(String fileExtension) {
+        return definitionFileFormat -> definitionFileFormat.getSupportedFileExtensions().contains(fileExtension);
+    }
```

Comment:
These two method are very similar with the ones in the other enum "DefinitionUploadType" only difference I can see is that one checks an array and one checks a string. Wonder would it be possible to  have a generic class with generic version of these methods... prob not worth the effort really

Reply:
Would that involve enum inheritance? 🤔 I think i get your point - I'm doubling up on the code I'm writing?

Reply:
Yeah there is a small bit of duplication,

You might be able todo something like this a the parent abstract class (Not sure if it has to be an Enum or not) 
...Maybe 🤔 

public static Optional<T> getEnumByValue(String value){
}

I wouldnt worry about it to much, just thinking out loud at this stage 😅